internal/crypto/nonce.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Stacklok, Inc.
```

Comment:
This code was previously in the same file as the crypto engine. Since it is unrelated, move to its own file.